src/redist/targets/GenerateBundledVersions.targets

```diff
@@ -324,6 +324,10 @@
       <WindowsDesktop31RuntimePackRids Include="@(WindowsDesktop30RuntimePackRids)" />
       <WindowsDesktop50RuntimePackRids Include="@(WindowsDesktop31RuntimePackRids);win-arm64" />
       <WindowsDesktopRuntimePackRids Include="@(WindowsDesktop50RuntimePackRids)" />
+
+      <_KnownRuntimeIdentiferPlatforms Include="$(ProductMonikerRid.Substring(0, $(Rid.LastIndexOf('-'))))" Condition="'$(DotNetBuildFromSource)' == 'true'" />
+      <_KnownRuntimeIdentiferPlatforms Include="any;freebsd;illumos;linux;linux-bionic;linux-musl;osx;solaris;unix;win" />
+      <_KnownRuntimeIdentiferPlatforms Include="android;aot;browser;ios;iossimulator;maccatalyst;tvos;tvossimulator" />
```

Comment:
Do we expect this list to change over time?  If the list might change for later .NET versions, it might be better to process `KnownFrameworkReference` / `KnownRuntimePack` items for the correct target framework to extract the platforms from the `RuntimePackRuntimeIdentifiers` metadata.

Reply:
It could if we add a new base/portable platform. Would it make sense to keep only the ones without packs (for example, `unix`) here and do the processing of `KnownFrameworkReference` / `KnownRuntimePack` for the corresponding target framework in Microsoft.NET.Sdk.targets when needed then?

Reply:
Updated to process the `KnownFrameworkReference`/`KnownRuntimePack` matching `$(TargetFramework)`. Generated text looks like:
```
<_KnownRuntimeIdentifersForTfm Include="@(KnownRuntimePack->WithMetadataValue('TargetFramework', $(TargetFramework))->Metadata('RuntimePackRuntimeIdentifiers'))" />
<_KnownRuntimeIdentifersForTfm Include="@(KnownFrameworkReference->WithMetadataValue('TargetFramework', $(TargetFramework))->Metadata('RuntimePackRuntimeIdentifiers'))" />
<_KnownRuntimeIdentifersForTfmWithPlatform Include="@(_KnownRuntimeIdentifersForTfm->ClearMetadata()->Distinct())">
  <Platform Condition="$([System.String]::new('%(Identity)').LastIndexOf('-')) == -1">%(Identity)</Platform>
  <Platform Condition="$([System.String]::new('%(Identity)').LastIndexOf('-')) != -1">$([System.String]::new('%(Identity)').Substring(0, $([System.String]::new('%(Identity)').LastIndexOf('-'))))</Platform>
</_KnownRuntimeIdentifersForTfmWithPlatform>

<_KnownRuntimeIdentiferPlatforms Include="any;aot;freebsd;illumos;solaris;unix" />
<_KnownRuntimeIdentiferPlatforms Include="@(_KnownRuntimeIdentifersForTfmWithPlatform->Metadata('Platform')->ClearMetadata()->Distinct())" Exclude="rhel.6;tizen.4.0.0;tizen.5.0.0" />
```

Is there a better way to do this?

Reply:
Moved the processing of the runtime packs to the SDK and kept the ones that don't have associated packs here. There's also the exclusion of the few platforms that have packs that we don't consider portable - since this can generate it only if those aren't the same platform as whatever is being used for a non-portable source-build.